src/ray/core_worker/context.h

```diff
@@ -104,7 +108,11 @@ class WorkerContext {
  private:
   const WorkerType worker_type_;
   const WorkerID worker_id_;
-  const JobID current_job_id_;
+
+  // a worker's job infomation might be lazily initialized.
+  JobID current_job_id_ GUARDED_BY(mutex_);
+  rpc::JobConfig job_config_ GUARDED_BY(mutex_);
```

Comment:
should we use optional here if it's lazily initialized?

Reply:
+1. Why don't we use optional instead of default config?